src/amo/components/OverallRating.js

```diff
@@ -70,12 +104,70 @@ export class OverallRatingBase extends React.Component {
   }
 }
 
-export const mapStateToProps = (state) => ({
-  apiState: state.api,
-  userId: state.auth && state.auth.userId,
-});
+export const mapStateToProps = (state, componentProps) => {
+  const userId = state.auth && state.auth.userId;
+  let userReview;
+
+  // Look for an already saved review by this user for this add-on/version.
+  if (userId && state.reviews) {
+    const allUserReviews = state.reviews[userId];
+    log.info(`Checking state for review by user ${userId},
+      addonId ${componentProps.addonId},
+      versionId ${componentProps.version.id}`);
+
+    if (allUserReviews) {
+      // TODO: adjust this when multiple reviews per version are stored
+      // in state.
+      // See https://github.com/mozilla/addons-server/issues/3986
+      const addonReview = allUserReviews[componentProps.addonId];
+      if (addonReview && addonReview.versionId === componentProps.version.id) {
+        // We have found an existing review by this user for this
+        // add-on and version.
+        userReview = addonReview;
+        log.info('Re-rendering component with user review from state',
+                 userReview);
+      }
+    }
+  }
+
+  return {
+    apiState: state.api,
+    userReview,
+    userId,
+  };
+};
 
 export const mapDispatchToProps = (dispatch) => ({
+
+  loadSavedRating({ addonId, versionId, userId }) {
+    return getUserReviews({ userId })
+      .then((response) => {
+        const relevantReviews = response.results.filter((review) => {
```

Comment:
Might be nice to move the filter into api and call it `getUserReview({ userId, addonId })` then. Paper over it until we can optimise it.